pkg/ebpf/c/tracee.bpf.c

```diff
@@ -1432,12 +1453,42 @@ static __always_inline int init_event_data(event_data_t *data, void *ctx)
     data->options = get_config(CONFIG_OPTIONS);
     init_context(&data->context, data->task, data->options);
     data->ctx = ctx;
-    data->buf_off = sizeof(context_t);
+    data->buf_off = sizeof(event_context_t);
     int buf_idx = SUBMIT_BUF_IDX;
     data->submit_p = bpf_map_lookup_elem(&bufs, &buf_idx);
     if (data->submit_p == NULL)
         return 0;
 
+    // try updating task_info_map with noexist flag. If successful, we will need to initialize it
+    // later. to save stack space (of size task_info_t) update the map with garbage data using the
+    // submit buffer.
+    int exist = bpf_map_update_elem(
+        &task_info_map, &data->context.task.host_tid, data->submit_p, BPF_NOEXIST);
+    data->task_info = bpf_map_lookup_elem(&task_info_map, &data->context.task.host_tid);
+    if (data->task_info == NULL) {
+        return 0;
+    }
+    if (!exist) {
+        data->task_info->syscall_traced = false;
+        data->task_info->new_task = false;
+        data->task_info->follow = false;
+        data->task_info->recompute_scope = true;
+    } else {
+        // check if we need to recompute scope due to context change
+        task_context_t *old_context = &data->task_info->context;
+        task_context_t *new_context = &data->context.task;
+
+        if ((old_context->cgroup_id == new_context->cgroup_id) &&
+            old_context->uid == new_context->uid && old_context->mnt_id == new_context->mnt_id &&
+            old_context->pid_id == new_context->pid_id &&
+            has_prefix(old_context->comm, new_context->comm, TASK_COMM_LEN) &&
+            has_prefix(old_context->uts_name, new_context->uts_name, TASK_COMM_LEN))
+            data->task_info->recompute_scope = true;
```

Comment:
Shouldn't this be false ? Thought: if nothing has changed, recompute_scope should be false then should_trace() wouldn't call do_should_trace() again (at least until context has changed).

Reply:
Wow, good catch.
I'll change this to '!=' and "||" instead